velox/type/Variant.cpp

```diff
@@ -555,7 +555,14 @@ uint64_t variant::hash() const {
       return folly::Hash{}(
           timestampValue.getSeconds(), timestampValue.getNanos());
     }
-
+    case TypeKind::SHORT_DECIMAL: {
+      auto shortDecimalValue = value<TypeKind::SHORT_DECIMAL>();
+      return folly::Hash{}(shortDecimalValue.unscaledValue.value());
```

Comment:
`shortDecimalValue.value()` should do.

Reply:
Done. Also, verified when this Hashing gets invoked. Here is the stack:
<img width="816" alt="Screen Shot 2022-08-02 at 11 38 09 AM" src="https://user-images.githubusercontent.com/6971561/182450364-f0135995-287f-4084-ae75-616f871c71f6.png">

Reply:
Hashing just the unscaled values is incorrect in this case. We must hash individual members of the DecimalCapsule variant object and return a combined hash value. We can follow similar to MAP typekind.